### PR TITLE
Fix compile error on FreeBSD

### PR DESCRIPTION
**Pull request by glowiak**
It causes a compile error, I have a patch to remove this anyway.

### CODE REVIEW

**Comment by SonyStinks** [comment]
Call me crazy, but since Java IS required, wouldn't it be better to simply A) detect Java, and B) if detection fails, just ask the user for the Java binary path?

**Comment by glowiak** [comment]
Well it is the check of java version, not the java itself. Before this code was added to launcher (around 1.4.x) everything was fine.

**Comment by LennyMcLennington** [comment]
This seems like the wrong way to fix a compiler error. What error does it cause? @glowiak

**Comment by slendidev** [comment]
This seems like a quick hack and not a good solution to the underlying issue here. What is the actual compile-time error?

**Comment by glowiak** [comment]
https://drive.google.com/file/d/1yMZ-kfTgUzW7aeXsIHgD4MvrPcQG-QC3/view?usp=share_link

**Comment by LennyMcLennington** [comment]
@glowiak what is that link? I can't view it. Can you just send the compiler error you get?

**Comment by glowiak** [comment]
It is a screenshot of terminal with this error because i can't copy text from xterm

**Comment by glowiak** [comment]
https://drive.google.com/file/d/1yMZ-kfTgUzW7aeXsIHgD4MvrPcQG-QC3/view?usp=share_link

**Comment by glowiak** [comment]
stupid google drive security this should work

**Comment by LennyMcLennington** [comment]
@glowiak it looks like it's happening due to FreeBSD defining `major` and `minor` macros... The more "correct" workaround would be to undefine those macros if they are present (or rename the major and minor identifiers to something else, like majorVersion or majorVer or something). Not to just omit any code that uses those identifiers.

**Comment by glowiak** [comment]
Yes, but the code for java checking is kinda complicated, I am not author of it and I do not wish to rename anything because I don't want to break anything.

**Comment by LennyMcLennington** [comment]
> Yes, but the code for java checking is kinda complicated, I am not author of it and I do not wish to rename anything because I don't want to break anything.

Then do the other thing I suggested and undef the macros that are defined by FreeBSD.

**Comment by glowiak** [comment]
No because something else would break

**Comment by LennyMcLennington** [comment]
> No because something else would break

Nothing in this codebase relies on the existence of some FreeBSD specific macros. The fact that the macros are defined is what breaks shit. Undefing them will unbreak shit.

**Comment by LennyMcLennington** [comment]
If you want to claim undefing them will break shit then why don't you try undefing them and prove that it breaks shit

**Comment by glowiak** [comment]
Appearantly the macro is already being undefined by some header present in the launcher, because such is contained by gnu coreutils too.

**Comment by LennyMcLennington** [comment]
> Appearantly the macro is already being undefined by some header present in the launcher, because such is contained by gnu coreutils too.

Well there you go then, just undefine it the same as it's done in the other header.

**Comment by glowiak** [comment]
what about now?

**Comment by LennyMcLennington** [comment]
> what about now?

If you can confirm this solves the issue I will merge it

**Comment by glowiak** [comment]
Right compile time

**Comment by glowiak** [comment]
@LennyMcLennington compiles and works perfectly

**Comment by glowiak** [comment]
@LennyMcLennington can you merge?

**Comment by glowiak** [comment]
@LennyMcLennington

File: launcher/minecraft/launch/VerifyJavaInstall.cpp [hunk 1]
```diff
@@ -39,6 +39,15 @@
 #include "minecraft/PackProfile.h"
 #include "minecraft/MinecraftInstance.h"
 
+#ifdef Q_OS_FREEBSD
+    #ifdef major
+        #undef major
+    #endif
+    #ifdef minor
+        #undef minor
+    #endif
```
Comment (slendidev) [inline launcher/minecraft/launch/VerifyJavaInstall.cpp]: You shouldn't need to `ifdef` for every macro. Just `undef` should do.

**Review by slendidev — CHANGES_REQUESTED** [review]

File: launcher/minecraft/launch/VerifyJavaInstall.cpp (hunk 1 above)
Comment (LennyMcLennington) [reply launcher/minecraft/launch/VerifyJavaInstall.cpp]: Also, the ifdef check for FreeBSD shouldn't be necessary. Any definition of macros with those names is likely to break things, it doesn't really need to only undef them specifically on FreeBSD

Comment (glowiak) [reply launcher/minecraft/launch/VerifyJavaInstall.cpp]: So here I go

**Comment by glowiak** [comment]
And yes you see that this won't work why don't undef it just on FBSD?

**Comment by glowiak** [comment]
@LennyMcLennington

**Pull request merged**